src/test/java/edu/carroll/cs389application/service/ImageServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.util.Pair;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for the ImageService implementation.
+ * <p>
+ * Combined pullimages and saveimage into one because they test all the functionality together.
+ */
+@SpringBootTest
+public class ImageServiceImplTest {
+
+    @Autowired
+    private ImageService imageService;
+
+    @Autowired
+    private UserService userService;
+
+    private final String testuser = "testuser";
+
+    private final LoginForm testform = new LoginForm(testuser);
+
+    /**
+     * Set up the mock objects before each test case.
+     *
+     * @throws IOException if there is an error in the setup process.
+     */
+    @BeforeEach
+    public void setup() throws IOException {
+        MockitoAnnotations.openMocks(this);
+        if (userService.validateUsername(testform)) {
+            //Create and validate our user in the DB.
+        }
+    }
+
+    /**
+     * This method will clean up the database after writing images.
+     *
+     * @throws IOException will throw if images failed to clean
+     */
+    @AfterEach
+    public void cleanup() throws IOException {
+        Login user = userService.loginFromUsername(testuser);
+        imageService.deleteUserImages(user);
+    }
+
+    /**
+     * Test saving a valid image file.
+     *
+     * @throws IOException if there is an error in the process of saving the image.
+     */
+    @Test
+    public void testSavePullImageOneValidFile() throws IOException {
+        Login user = userService.loginFromUsername(testuser);
+        BufferedImage image = new BufferedImage(10, 10, BufferedImage.TYPE_BYTE_BINARY);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(image, "png", baos);
+        byte[] fileBytes = baos.toByteArray();
+        MultipartFile file1 = new MockMultipartFile("test-image1.jpg", "image1.jpg", "image/jpeg", fileBytes);
+        ImageForm form1 = new ImageForm(file1);
+        imageService.saveImage(form1, user);
+        List<Pair<InputStream, String>> images = imageService.pullImages(user);
+        List<UserImage> userimages = imageService.loadUserImagesbyUserID(user);
+        for (Pair<InputStream, String> pair : images) {
+            byte[] pairbytes = pair.getFirst().readAllBytes();
+            assertArrayEquals(pairbytes, fileBytes);
+        }
+        String contentType1 = userimages.get(0).getExtension().equalsIgnoreCase("jpg") || userimages.get(0).getExtension().equalsIgnoreCase("jpeg") ? "image/jpeg" : "image/png";
```

Comment:
You assume there is at least one image here.  What would happen if there were more than 1 image returned?  Would it be an error?  Would your test pass in that case, and should it?